### ISSUE TITLE
DIAMOND training guide and expected results

### ISSUE THREAD

**Issue by ParamThakkar123** [ParamThakkar123/torchwm#186]
## Summary

Add documentation covering how to train DIAMOND, expected rewards per game, and links to pretrained checkpoints. Low code but high value for users trying to reproduce results.

### What to document

1. **Training commands** — examples for each preset:
   ```
   torchwm train diamond --game Breakout-v5 --seed 0 --preset medium
   torchwm train diamond --game Pong-v5 --seed 0 --preset small
   ```

2. **Expected results** — table of rewards after N epochs per game using the standard Atari 100k protocol. Reference the DIAMOND paper numbers and note any differences.

3. **Checkpoint format** — what's saved in a checkpoint (config, diffusion weights, reward model, actor-critic, optimizer states, replay buffer)

4. **Resume training** — how to resume from a checkpoint

5. **Evaluation commands** — 
   ```
   torchwm eval --model diamond --checkpoint checkpoint.pt --game Breakout-v5
   torchwm play --model diamond --checkpoint checkpoint.pt --game Breakout-v5
   ```

### Where to add
- `docs/source/diamond.md` or the existing `docs/` structure
- Also add a brief section to the repo README

### Optional
- Link to pretrained checkpoints (HuggingFace or GitHub releases)
- Training curves (screenshots or plotly figures)

## Acceptance criteria

- [ ] Training commands documented with all presets
- [ ] Expected reward table for 3+ Atari games
- [ ] Eval and play commands documented
- [ ] Checkpoint format explained
- [ ] Resume training instructions
